### ISSUE TITLE
"Invalid column index" on Oracle stored procedure call when return type is converted to boolean and argument type is defaulted

### ISSUE THREAD

**Issue by dsharinov** [jOOQ/jOOQ#6498]
jOOQ 3.9.5
The problem happens on stored procedure with default parameters that return 'Y'/'N' value. We have a simple converter to translate 'Y'/'N' values to Java Boolean type:
```
                    forcedType {
                        userType = 'java.lang.Boolean'
                        converter = 'com.company.converters.BooleanConverter'
                        types = '(CHAR|VARCHAR2)\\(1(\\s|\\))'
                    }
```
I've attached source code for converter and test Oracle package. The test itself is very simple:
```
    @Autowired
    private DSLContext jooq;

    @Test
    public void testBoolParam() {
        Getflag getflag = new Getflag();
        //getflag.setFlag(false); // DO NOT SET THE PARAMETER!
        getflag.execute(jooq.configuration());
    }
```
It fails with "Invalid column index". jOOQ decides that it needs to generate PL/SQL block in _AbstractRoutine#toSQLAssign_:
```
        if (needsPLSQLBlock() && context.family() == ORACLE) {
            if (getReturnParameter().getType() == Boolean.class)
                context.sql("b1 := ");
```
and then in _AbstractRoutine#bind1_ 
```
        // Skip one index for OUT parameters
        else {
            context.nextIndex();
        }
```
the parameter index is increased, but Oracle sees only one parameter to bind.
[TestBool.zip](https://github.com/jOOQ/jOOQ/files/1223390/TestBool.zip)

**Comment by lukaseder** [comment]
Thank you very much for your report. We'll look into this ASAP

**Comment by lukaseder** [comment]
Thank you very much for your patience. Indeed, I can reproduce this problem. The workaround is obviuosly to pass the parameter with the call, but this should work without the parameter as well.

**Comment by lukaseder** [comment]
That line in `toSQLAssign()` should read:

```java
        if (needsPLSQLBlock() && context.family() == ORACLE) {
            if (booleanParameter(getReturnParameter()))
                context.sql("b1 := ");
```

This will be fixed in 3.10, 3.9.6, 3.8.9, all three releases due for next week, before the end of Sept 2017
